libpirate/libpirate.h

```diff
@@ -286,6 +286,18 @@ int pirate_parse_channel_param(const char *str, pirate_channel_param_t *param);
 
 int pirate_get_channel_param(int gd, pirate_channel_param_t *param);
 
+// Get channel configuration
+//
+// Parameters
+//  gd           - GAPS channel number
+//  desc         - string to contain channel description
+//  len          - max len allowed in desc
+//
+// Return:
+//  0 on success
+// -1 on failure, errno is set
+int pirate_get_channel_description(int gd, char *desc, int len);
```

Comment:
Can you change the behavior of the return value to match snprintf() and update the documentation:

```Upon  successful  return, this function returns the number of characters printed (excluding the null byte used to end output to strings). The function does not write more than len bytes (including the terminating null byte ('\0')). If the output was truncated due to this limit, then the return value is the number of characters (excluding the terminating null byte) which would have been written to the final string if enough space had been available. Thus, a return value of len or more means that the output was truncated. If an output error is encountered, a negative value is returned.```

Reply:
Done